### PR TITLE
Support PoL heroes & artifacts in "battle only" mode

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
Currently heroes from PoL cannot be used in "battle only" mode (they turn into Heroes::UNKNOWN) and I suppose that there may be also issues with artifacts from PoL if this addon is not available on player's machine.

Need a review from someone who is familiar with PoL addon mechanics to make sure that I do the right things here and didn't forget anything. PoL doesn't add new secondary skills, spells or monsters, right?

### CODE REVIEW

**Comment by oleg-derevenetz** [comment]
Not sure is this really a regression though...

**Comment by ihhub** [comment]
> Currently heroes from PoL cannot be used in "battle only" mode (they turn into Heroes::UNKNOWN) and I suppose that there may be also issues with artifacts from PoL if this addon is not available on player's machine.
> 
> Need a review from someone who is familiar with PoL addon mechanics to make sure that I do the right things here and didn't forget anything. PoL doesn't add new secondary skills, spells or monsters, right?

Hi @oleg-derevenetz , PoL doesn't adds only new artifacts and buildings on the map.

**Review by ihhub — CHANGES_REQUESTED** [review]

File: src/fheroes2/world/world.cpp [hunk 1]
```diff
@@ -300,12 +300,26 @@ void World::Reset( void )
...
+
+    Maps::FileInfo fi;
+
+    fi.size_w = static_cast<uint16_t>( width );
+    fi.size_h = static_cast<uint16_t>( height );
+
+    Settings & conf = Settings::Get();
+
+    if ( conf.isPriceOfLoyaltySupported() ) {
+        fi._version = GameVersion::PRICE_OF_LOYALTY;
```
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/world/world.cpp:315]: Hi @ihhub As far as I understand, this is the special case of auto-generated "map" created for "battle only" mode just to have some "world". This function is called only for `TYPE_BATTLEONLY` game type.

[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/world/world.cpp:315]: BTW it seems that all this auto-generated 10x10 map consists of "water" tiles, that's why "battle only" battles are always on board of the ship.

[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/world/world.cpp:315]: Ah, I misread it.

[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/world/world.cpp:315]: For water tile: you have to modify `mp2tile.tileIndex = static_cast<uint16_t>( Rand::Get( 16, 19 ) );` line to enable other types of surfaces.

[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/world/world.cpp:315]: @ihhub

> For water tile: you have to modify mp2tile.tileIndex = static_cast<uint16_t>( Rand::Get( 16, 19 ) ); line to enable other types of surfaces.

Yes, I figured it out. Boarding battles are OK for me :)

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you very much for improving the game!

**Pull request merged**